python_modules/dagster/dagster/_core/definitions/metadata/__init__.py

```diff
@@ -614,129 +615,124 @@ def null() -> "NullMetadataValue":
 
 
 @whitelist_for_serdes(storage_name="TextMetadataEntryData")
-class TextMetadataValue(
-    NamedTuple(
-        "_TextMetadataValue",
-        [
-            ("text", PublicAttr[Optional[str]]),
-        ],
-    ),
-    MetadataValue[str],
-):
+class TextMetadataValue(DagsterModel, MetadataValue[str]):
     """Container class for text metadata entry data.
 
     Args:
         text (Optional[str]): The text data.
     """
 
-    def __new__(cls, text: Optional[str]):
-        return super(TextMetadataValue, cls).__new__(
-            cls, check.opt_str_param(text, "text", default="")
-        )
+    text_inner: Optional[str] = Field(..., alias="text")
+
+    def __init__(self, text: Optional[str]):
+        super().__init__(text=text or "")
```

Comment:
^ or we could just drop this custom `__init__` and let the `None` flow through since thats how everything is typed

Reply:
I agree that this is weird - this was just me trying to minimize changes to the existing behavior.

Reply:
got it - consider dropping the unnecessary `Optional` in a follow up PR then I guess. Hard to argue that its that impactful of an improvement